org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/ProgressReporterManager.java

```diff
@@ -160,14 +166,29 @@ protected void sendStatus() {
 			progressReport.setTotalWork(totalWork);
 			progressReport.setWorkDone(progress);
 			progressReport.setComplete(isDone());
-			String subTask = subTaskName == null ? "" : " - " + subTaskName;
-			progressReport.setStatus(formatMessage(task + subTask));
+			if (task != null && subTaskName != null && !subTaskName.isEmpty() && task.equals(MavenProjectImporter.IMPORTING_MAVEN_PROJECTS)) {
+				progressReport.setStatus(task + SEPARATOR + subTaskName);
+			} else {
+				progressReport.setStatus(formatMessage(task));
+			}
 			client.sendProgressReport(progressReport);
 		}
 
 
 		protected String formatMessage(String task) {
-			return (totalWork > 0) ? String.format("%s - %.0f%%", task, ((double) progress / totalWork) * 100) : task;
+			String message = getMessage(task);
+			return (totalWork > 0) ? String.format("%.0f%%%s", ((double) progress / totalWork) * 100, message) : message;
```

Comment:
a space is missing between the percentage and the message

Reply:
getMessage adds a space. You can try ProgressReporterManagerTest.testStartupJobReporting()

Reply:
Check progress reports like "0%Building workspace"